TrackletAlgorithm/InputRouter.h

```diff
@@ -238,7 +237,7 @@ void InputRouter( const BXType bx
 	ClearCounters<nOMems>(nMems,  hNStubs);
 
 	LOOP_ProcessIR:
-	for (int cStubCounter = 0; cStubCounter < kMaxStubsFromLink; cStubCounter++) 
+	for (int cStubCounter = 0; cStubCounter < kMaxProc; cStubCounter++) 
```

Comment:
How does  the code fail without this change? Naively the VHDL produced by the HLS compiler shouldn't depend on the assumed length of the FIFO (or equivalently on the number of loops per TM period).

Reply:
It fails because the IR sends out the done signal every 256 (kMaxStubsFromLink) clocks even if the input bx changes every 108 clocks.